tests/AuthenticationBundle/RelationshipVoter/SchoolTest.php

```diff
@@ -0,0 +1,120 @@
+    }
+
+    public function testCanNotView()
+    {
+        $token = $this->createMockTokenWithNonRootSessionUser();
+        $entity = m::mock(School::class);
+        $entity->shouldReceive('getId')->andReturn(1);
+        $entity->shouldReceive('getSchool')->andReturn(1);
+        $this->permissionChecker->shouldReceive('canReadSchool')->andReturn(false);
+        $response = $this->voter->vote($token, $entity, [AbstractVoter::VIEW]);
+        $this->assertEquals(VoterInterface::ACCESS_DENIED, $response, " View denied");
+    }
+
+    public function testCanEdit()
+    {
+        $token = $this->createMockTokenWithNonRootSessionUser();
+        $entity = m::mock(School::class);
+        $entity->shouldReceive('getId')->andReturn(1);
+        $entity->shouldReceive('getSchool')->andReturn(1);
+        $this->permissionChecker->shouldReceive('canUpdateSchool')->andReturn(true);
+        $response = $this->voter->vote($token, $entity, [AbstractVoter::EDIT]);
+        $this->assertEquals(VoterInterface::ACCESS_GRANTED, $response, " Edit allowed");
+    }
+
+    public function testCanNotEdit()
+    {
+        $token = $this->createMockTokenWithNonRootSessionUser();
+        $entity = m::mock(School::class);
+        $entity->shouldReceive('getId')->andReturn(1);
+        $entity->shouldReceive('getSchool')->andReturn(1);
+        $this->permissionChecker->shouldReceive('canUpdateSchool')->andReturn(false);
+        $response = $this->voter->vote($token, $entity, [AbstractVoter::EDIT]);
+        $this->assertEquals(VoterInterface::ACCESS_DENIED, $response, " Edit denied");
+    }
+
+    public function testCanDelete()
+    {
+        $token = $this->createMockTokenWithNonRootSessionUser();
+        $entity = m::mock(School::class);
+        $entity->shouldReceive('getId')->andReturn(1);
+        $entity->shouldReceive('getSchool')->andReturn(1);
+        $this->permissionChecker->shouldReceive('canDeleteSchool')->andReturn(true);
+        $response = $this->voter->vote($token, $entity, [AbstractVoter::DELETE]);
+        $this->assertEquals(VoterInterface::ACCESS_GRANTED, $response, " Delete allowed");
+    }
+
+    public function testCanNotDelete()
+    {
+        $token = $this->createMockTokenWithNonRootSessionUser();
+        $entity = m::mock(School::class);
+        $entity->shouldReceive('getId')->andReturn(1);
+        $entity->shouldReceive('getSchool')->andReturn(1);
+        $this->permissionChecker->shouldReceive('canDeleteSchool')->andReturn(false);
+        $response = $this->voter->vote($token, $entity, [AbstractVoter::DELETE]);
+        $this->assertEquals(VoterInterface::ACCESS_DENIED, $response, " Delete allowed");
+    }
+
+    public function testCanNotCreate()
+    {
+        $token = $this->createMockTokenWithNonRootSessionUser();
+        $entity = m::mock(School::class);
+        $entity->shouldReceive('getId')->andReturn(1);
+        $this->permissionChecker->shouldReceive('canCreateSchool')->andReturn(false);
```

Comment:
This test shouldn't pass because `canCreateSchool` is never called. It does pass so I'm guessing we need to add `->once()` to all these expectations to make sure we're testing the right thing.

Reply:
good catch, i removed that code.